tests/json_infra/helpers/load_blockchain_tests.py

```diff
@@ -133,96 +58,150 @@ def add_block_to_chain(
-                _identifier = (
-                    "("
-                    + _test_case["test_file"]
-                    + "|"
-                    + _test_case["test_key"]
-                    + ")"
-                )
-                _test_case["eels_fork"] = eels_fork
-                if any(
-                    x.search(_identifier) for x in test_patterns.expected_fail
-                ):
-                    continue
-                elif any(x.search(_identifier) for x in test_patterns.slow):
-                    yield pytest.param(_test_case, marks=pytest.mark.slow)
-                elif any(
-                    x.search(_identifier) for x in test_patterns.big_memory
-                ):
-                    yield pytest.param(_test_case, marks=pytest.mark.bigmem)
-                else:
-                    yield _test_case
-        except NoTestsFoundError:
-            # file doesn't contain tests for the given fork
-            continue
-
-
-# Test case Identifier
-def idfn(test_case: Dict) -> str:
-    """Generate test case identifier from test case dictionary."""
-    if isinstance(test_case, dict):
-        folder_name = test_case["test_file"].split("/")[-2]
-        # Assign Folder name and test_key to identify tests in output
-        return folder_name + " - " + test_case["test_key"]
+            state = load.json_to_state(json_data["pre"])
+        except StateWithEmptyAccount as e:
+            pytest.xfail(str(e))
+
+        chain = load.fork.BlockChain(
+            blocks=[genesis_block],
+            state=state,
+            chain_id=U64(json_data["genesisBlockHeader"].get("chainId", 1)),
+        )
+
+        mock_pow = (
+            json_data["sealEngine"] == "NoProof"
+            and not load.fork.proof_of_stake
+        )
+
+        for json_block in json_data["blocks"]:
+            block_exception = None
+            for key, value in json_block.items():
+                if key.startswith("expectException"):
+                    block_exception = value
+                    break
+                if key == "exceptions":
+                    block_exception = value
+                    break
+
+            if block_exception:
+                # TODO: Once all the specific exception types are thrown,
+                #       only `pytest.raises` the correct exception type instead
+                #       of all of them.
+                with pytest.raises((EthereumException, RLPException)):
+                    add_block_to_chain(chain, json_block, load, mock_pow)
+                return
+            else:
+                add_block_to_chain(chain, json_block, load, mock_pow)
+
+        last_block_hash = hex_to_bytes(json_data["lastblockhash"])
+        assert (
+            keccak256(rlp.encode(chain.blocks[-1].header)) == last_block_hash
+        )
+
+        expected_post_state = load.json_to_state(json_data["postState"])
+        assert chain.state == expected_post_state
```

Comment:
I think this is currently not set up to catch any tests where the blocks themselves do not throw any exceptions but the overall state comparison fails . This I think is causing the current CI failure